examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -440,16 +485,59 @@ def _cumulated_claims(y_true, y_pred, exposure):
 ax.legend(loc="upper left")
 
 ##############################################################################
-# This plot reveals that the random forest model is slightly better at ranking
-# policyholders by risk profiles even if the absolute value of the predicted
-# expected frequencies are less well calibrated than for the linear Poisson
-# model.
+# As expected, the dummy regressor is unable to discriminate and therefore
+# performs the worst on this plot.
+#
+# The tree-based model is significantly better at ranking policyholders by risk
+# while the two linear models perform similarly.
 #
 # All three models are significantly better than chance but also very far from
 # making perfect predictions.
 #
 # This last point is expected due to the nature of the problem: the occurrence
 # of accidents is mostly dominated by circumstantial causes that are not
 # captured in the columns of the dataset or that are indeed random.
+#
+# The linear models assume no interactions between the input variables which
+# likely causes under-fitting. Inserting a polynomial feature extractor
+# (func:`sklearn.preprocessing.PolynomialFeatures`) indeed increases their
+# discrimative power by 2 points of Gini index. In particular it improves the
+# ability of the models to identify the top 5% riskiest profiles.
+#
+# Main takeaways
+# --------------
+#
+# - A ideal model is both well-calibrated and discriminative.
+#
+# - The Gini index reflects the ability of a model to rank predictions
+#   irrespective of their absolute values, and therefore only assess their
+#   discriminative power.
+#
+# - The calibration of the model can be assessed by plotting the mean observed
+#   value vs the mean predicted value on groups of test samples binned by
+#   predicted risk.
+#
+# - The least squares loss of the Ridge regression model seem to cause this
+#   model to be badly calibrated. In particular it tends to under estimate the
+#   risk and can even predict invalid negative frequencies...
+#
+# - Using the Poisson loss can correct this problem and lead to a
+#   well-calibrated linear model.
+#
+# - Despite the improvement in calibration, the discriminative power of both
+#   linear models are comparable and well below the discriminative power of the
+#   Gradient Boosting Regression Trees.
+#
+# - The non-linear Gradient Boosting Regression Trees model does not seem to
+#   suffer from significant mis-calibration issues (despite the use of a least
+#   squares loss).
+#
+# - The Poisson deviance computed as an evaluation metric reflects both the
+#   calibration and the discriminative power of the model but makes a linear
```

Comment:
```suggestion
#   calibration and the discriminative power of the model and makes a linear
```
The squared error stemming from the normal distribution also makes an assumption on the variance.